### ISSUE TITLE
Intermittent failure - `koa-body` attaches empty object

### ISSUE THREAD

**Issue by dylanpyle** [dlau/koa-body#45]
About 50% of my POST requests result in `koa-body` attaching an empty object (i.e `this.request.body = {}`), as opposed to a set of files uploaded.

Here's a request that failed:
[headers](https://github.com/dylanpyle/debugging-koa-body/blob/master/failed-request-headers.txt)
[body](https://github.com/dylanpyle/debugging-koa-body/blob/master/failed-request-body.bin)

And one that succeeded:
[headers](https://github.com/dylanpyle/debugging-koa-body/blob/master/succeeded-request-headers.txt)
[body](https://github.com/dylanpyle/debugging-koa-body/blob/master/succeeded-request-body.bin)

I can't see anything notably different between these. Digging into Formidable now — I assume there's a decent chance it's coming from there — but figured it was worth documenting here.

**Comment by dylanpyle** [comment]
Seems like `koa-better-body` is exhibiting the same behavior. Sorry for the noise.

**Comment by dylanpyle** [comment]
For posterity: I never managed to track down the issue here. All `koa` middlewares (koa-better-body, multer, etc) seem to exhibit the same behavior, but other non-node servers I tested did not.

The only _client_ I found that caused this specific failure mode was the JavaScript XHR / fetch API in React Native. I haven't figured out whether the misbehavior here is on the client or server, but I've managed to avoid it by switching my client to use [RNFetchBlob](https://github.com/wkh237/react-native-fetch-blob) instead.
